Cubical/Core/Prelude.agda

```diff
@@ -78,12 +78,58 @@ compPathP : {A : I → Set ℓ} → {x : A i0} → {y : A i1} → {B_i1 : Set 
   (p : PathP A x y) → (q : PathP (λ i → B i) y z) → PathP (λ j → ((λ i → A i) ∙ B) j) x z
 compPathP p q j = compPathP-filler p q j i1
 
+∙-rInv : (p : x ≡ y) → p ∙ (sym p) ≡ refl
+∙-rInv {x = x} p i j = hcomp (λ k → \ { (i = i0) → compPath-filler p (sym p) k j
+                              ; (i = i1) → p (j ∧ (~ k))
+                              ; (j = i0) → x
+                              ; (j = i1) → p (~ k) })
+                     (p j)
+
+∙-rUnit : (p : x ≡ y) → p ∙ refl ≡ p
+∙-rUnit {x = x} {y = y} p i j =
+  hcomp (λ k → \ { (i = i0) → compPath-filler p refl k j
+                 ; (i = i1) → p j
+                 ; (j = i0) → x
+                 ; (j = i1) → y }) (p j)
+
 _≡⟨_⟩_ : (x : A) → x ≡ y → y ≡ z → x ≡ z
 _ ≡⟨ x≡y ⟩ y≡z = x≡y ∙ y≡z
 
 _∎ : (x : A) → x ≡ x
 _ ∎ = refl
 
+-- another definition of composition, useful for some proofs
```

Comment:
Interesting, can you point me to some proofs where this is useful?

I think this should be defined using the triple composition of @loic-p :  https://github.com/agda/cubical/blob/master/Cubical/Foundations/GroupoidLaws.agda#L179

Maybe the same holds for the normal compPath (now called \.)? In fact, maybe we can develop some lemmas for the triple composition and then recover the lemmas for compPath and compPath' from the general one?

Reply:
yes, that sounds a good idea. In fact, even better, we can define compPath x y as tripleComp refl x y, and compPath' x y as tripleComp x y refl. I will work on this.